src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -294,12 +297,38 @@ private void SetGasPerBlock(ApplicationEngine engine, BigInteger gasPerBlock)
         /// </summary>
         /// <param name="snapshot">The snapshot used to read data.</param>
         /// <returns>The amount of GAS generated.</returns>
+        [Obsolete("Use GetGasPerBlock(DataCache) instead")]
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
         public BigInteger GetGasPerBlock(DataCache snapshot)
         {
             return GetSortedGasRecords(snapshot, Ledger.CurrentIndex(snapshot) + 1).First().GasPerBlock;
         }
 
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates | CallFlags.WriteStates)]
+        public BigInteger GetGasPerBlock(ApplicationEngine engine, DataCache snapshot)
+        {
+            var currentIndex = Ledger.CurrentIndex(snapshot) + 1;
+            var currentGasPerBlock = GetSortedGasRecords(snapshot, currentIndex).First().GasPerBlock;
+
+            if (!engine.IsHardforkEnabled(Hardfork.HF_Echidna))
+                return currentGasPerBlock;
+
+            var hardforkHeight = engine.ProtocolSettings.Hardforks[Hardfork.HF_Echidna];
+
+            if (hardforkHeight != currentIndex)
+            {
+                return currentGasPerBlock;
+            }
+
+            // At the height of the hardfork Echidna, the gas per block will be reduced by 80%
+            // because the block time is reduced by 80%.
+            currentGasPerBlock /= 5; // I think it should not be a problem for this code to ensure the result is not 0
```

Comment:
NACK. This MUST NOT be done at the code level. We have committee to do this and different networks can legitimately have very different settings.

Reply:
Sure, you and shargon share the same idea, if you think gasperblock should be a pure council thing, i have no problem with that at all, lets discuss it in the meeting.

Reply:
We can avoid the reward if is higer than X, this will push committee to do the change

Reply:
according to roman's comment, we should not touch the GAS reward at all, so i assume maybe that allow a few block to have 5 gas minted after the hardfork is acceptable. Lets discuss it in the meeting.